psi4/src/psi4/ccenergy/ccenergy.cc

```diff
@@ -654,4 +860,444 @@ void CCEnergyWavefunction::one_step(void) {
+    global_dpd_->file2_init(&fIJ, PSIF_CC_OEI, 0, 0, 0, "fIJ");
+    global_dpd_->file2_init(&fij, PSIF_CC_OEI, 0, 0, 0, "fij");
+    global_dpd_->file2_init(&fAB, PSIF_CC_OEI, 0, 1, 1, "fAB");
+    global_dpd_->file2_init(&fab, PSIF_CC_OEI, 0, 1, 1, "fab");
+    global_dpd_->file2_init(&fIA, PSIF_CC_OEI, 0, 0, 1, "fIA");
+    global_dpd_->file2_init(&fia, PSIF_CC_OEI, 0, 0, 1, "fia");
+    // ...copy them in the "original" file...
+    global_dpd_->file2_copy(&fIJ, PSIF_CC_OEI, "fIJ original");
+    global_dpd_->file2_copy(&fij, PSIF_CC_OEI, "fij original");
+    global_dpd_->file2_copy(&fAB, PSIF_CC_OEI, "fAB original");
+    global_dpd_->file2_copy(&fab, PSIF_CC_OEI, "fab original");
+    global_dpd_->file2_copy(&fIA, PSIF_CC_OEI, "fIA original");
+    global_dpd_->file2_copy(&fia, PSIF_CC_OEI, "fia original");
+    // ...and close.
+    global_dpd_->file2_close(&fIJ);
+    global_dpd_->file2_close(&fij);
+    global_dpd_->file2_close(&fAB);
+    global_dpd_->file2_close(&fab);
+    global_dpd_->file2_close(&fIA);
+    global_dpd_->file2_close(&fia);
+  }
+  // Read from the original Fock matrix (the one coming from the PCM-SCF step)
+  global_dpd_->file2_init(&fIJ, PSIF_CC_OEI, 0, 0, 0, "fIJ original");
+  global_dpd_->file2_mat_init(&fIJ);
+  global_dpd_->file2_mat_rd(&fIJ);
+  global_dpd_->file2_init(&fij, PSIF_CC_OEI, 0, 0, 0, "fij original");
+  global_dpd_->file2_mat_init(&fij);
+  global_dpd_->file2_mat_rd(&fij);
+  global_dpd_->file2_init(&fAB, PSIF_CC_OEI, 0, 1, 1, "fAB original");
+  global_dpd_->file2_mat_init(&fAB);
+  global_dpd_->file2_mat_rd(&fAB);
+  global_dpd_->file2_init(&fab, PSIF_CC_OEI, 0, 1, 1, "fab original");
+  global_dpd_->file2_mat_init(&fab);
+  global_dpd_->file2_mat_rd(&fab);
+  global_dpd_->file2_init(&fIA, PSIF_CC_OEI, 0, 0, 1, "fIA original");
+  global_dpd_->file2_mat_init(&fIA);
+  global_dpd_->file2_mat_rd(&fIA);
+  global_dpd_->file2_init(&fia, PSIF_CC_OEI, 0, 0, 1, "fia original");
+  global_dpd_->file2_mat_init(&fia);
+  global_dpd_->file2_mat_rd(&fia);
+  // Open a series of buffers to put the various blocks in
+  dpdfile2 fIJ2, fij2, fAB2, fab2, fIA2, fia2;
+  global_dpd_->file2_init(&fIJ2, PSIF_CC_OEI, 0, 0, 0, "fIJ");
+  global_dpd_->file2_mat_init(&fIJ2);
+  global_dpd_->file2_mat_rd(&fIJ2);
+  global_dpd_->file2_init(&fij2, PSIF_CC_OEI, 0, 0, 0, "fij");
+  global_dpd_->file2_mat_init(&fij2);
+  global_dpd_->file2_mat_rd(&fij2);
+  global_dpd_->file2_init(&fAB2, PSIF_CC_OEI, 0, 1, 1, "fAB");
+  global_dpd_->file2_mat_init(&fAB2);
+  global_dpd_->file2_mat_rd(&fAB2);
+  global_dpd_->file2_init(&fab2, PSIF_CC_OEI, 0, 1, 1, "fab");
+  global_dpd_->file2_mat_init(&fab2);
+  global_dpd_->file2_mat_rd(&fab2);
+  global_dpd_->file2_init(&fIA2, PSIF_CC_OEI, 0, 0, 1, "fIA");
+  global_dpd_->file2_mat_init(&fIA2);
+  global_dpd_->file2_mat_rd(&fIA2);
+  global_dpd_->file2_init(&fia2, PSIF_CC_OEI, 0, 0, 1, "fia");
+  global_dpd_->file2_mat_init(&fia2);
+  global_dpd_->file2_mat_rd(&fia2);
+
```

Comment:
From reading below you might want to add a function that can write a Matrix to a `dpdfile` and then use Matrix operations for all of the below code since its just `AXPY`'s and our new slicing notation. See [here](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libscf_solver/rohf.cc#L1075).